library/src/main/java/com/novoda/downloadmanager/lib/DownloadBatch.java

```diff
@@ -125,4 +149,77 @@ public boolean scanCompletedMediaIfReady(DownloadScanner downloadScanner) {
         return false;
     }
 
+    public boolean shouldShowActiveItem() {
+        int visibility = info.getVisibility();
+        return visibility == NotificationVisibility.ONLY_WHEN_ACTIVE
+                || visibility == NotificationVisibility.ACTIVE_OR_COMPLETE;
+    }
+
+    public boolean shouldShowCompletedItem() {
+        int visibility = info.getVisibility();
+        return visibility == NotificationVisibility.ONLY_WHEN_COMPLETE
+                || visibility == NotificationVisibility.ACTIVE_OR_COMPLETE;
+    }
+
+    public String getBigPictureUrl() {
+        return info.getBigPictureUrl();
+    }
+
+    public String getDescription() {
+        return info.getDescription();
+    }
+
+    public String getTitle() {
+        return info.getTitle();
+    }
+
+    public long getFirstDownloadBatchId() {
+        return downloads.get(0).getId();
+    }
+
+    public Statistics getLiveStatistics(LongSparseArray<Long> downloadSpeed) {
+        long currentBytes = 0;
```

Comment:
Also, did you create this code or move this code? This will result in a ton of allocations D: